src/common/buf.c

```diff
@@ -2,25 +2,23 @@
 #include <assert.h>
 #include <ctype.h>
 #include <stdbool.h>
+#include <string.h>
 #include "common/buf.h"
+#include "common/macros.h"
 #include "common/mem.h"
 
 void
 buf_expand_tilde(struct buf *s)
 {
-	struct buf new;
-	buf_init(&new);
+	struct buf new = BUF_INIT;
```

Comment:
This feels a bit weird, what about `struct buf new = BUF_INIT()` instead?

Reply:
I don't have a strong preference, but the parentheses seem unnecessary to me since it's just a predefined value, not a functional macro.

Reply:
A define "returning" a struct feels different to a macro returning a struct.

Reply:
I'm not following. It's not "returning" anything, I don't think? It's just a literal struct value.

Reply:
Yes,but usually a define defines a simple value (some number or string) and not a struct initializer.

The way I see it: if you imagine there is no `#define` involved at all you wouldn't see code like this, e.g. initializing a struct (not a struct pointer like `struct foo *bar`) by assigning it a single variable. Instead you would have a function that returns a struct which is then copied over to the new instance.

To mimic that usual behavior I think having `BUF_INIT()` here would be way easier to read, even without knowing what the `BUF_INIT` define / macro is actually doing.

Reply:
Code assigning/initializing a whole struct at once is legal in C even if uncommon -- it's more common in other languages maybe.

Anyway this seems like a minor detail. It's not a big deal to me either way.

Reply:
I prefer `BUF_INIT` and it similar to `git`'s approach.